### PR TITLE
Rewrite script which generates regressions tests

### LINKED ISSUE #3213: Investigate failing test update job

### PR DESCRIPTION
**Pull request by ana-pantilie**
Fixes #3213 

- new script generates bug reports from downstream (currently just KEVM) tests
- the bug reports are modified to comply with current Makefile testing harness
- removes tests which are no longer supported by downstream project

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: scripts/generate-regression-tests.sh [hunk 1]
```diff
@@ -5,20 +5,32 @@ set -exuo pipefail
...
+    mv $archive $tmp
+    cd $tmp
+    tar -xf $archive
+    rm ./!(kore-exec.sh|spec.kore|vdefinition.kore)
+    mv kore-exec.sh $script
+    mv vdefinition.kore $def
+    mv spec.kore $spec
+
+    $KORE/scripts/trim-source-paths.sh *.kore
+    gsed -i "s/result.kore/$script/g" test-$name.sh
```
Comment (ana-pantilie) [inline scripts/generate-regression-tests.sh:27]:
```suggestion
    gsed -i "s/result.kore/$script.out/g" test-$name.sh
```

**Review by goodlyrottenapple — APPROVED** [review]
LGTM

File: scripts/generate-regression-tests.sh [hunk 2]
```diff
@@ -5,20 +5,32 @@ set -exuo pipefail
...
+    tar -xf $archive
+    rm ./!(kore-exec.sh|spec.kore|vdefinition.kore)
+    mv kore-exec.sh $script
+    mv vdefinition.kore $def
+    mv spec.kore $spec
+
+    $KORE/scripts/trim-source-paths.sh *.kore
+    gsed -i "s/result.kore/$script.out/g" test-$name.sh
+    gsed -i "s/vdefinition.kore/$def/g" test-$name.sh
+    gsed -i "s/spec.kore/$spec/g" test-$name.sh
```
Comment (ana-pantilie) [inline scripts/generate-regression-tests.sh:29]:
```suggestion
    sed -i "s/result.kore/$script.out/g" test-$name.sh
    sed -i "s/vdefinition.kore/$def/g" test-$name.sh
    sed -i "s/spec.kore/$spec/g" test-$name.sh
```

**Comment by ehildenb** [comment]
Somehow @goodlyrottenapple 's Nix script for running profiles (https://github.com/runtimeverification/haskell-backend/pull/3235) already gives a way to take a tarball and run a profile for it via the Nix build system.

Can there be a way we just re-use that a bit here, but instead of turning on profiling we collect and inspect the output? That way we just have to regenertae and store the new tarballs, and it's using the same infrastructure for profiling or for downsteram testing?

Not sure it makes sense, but I figure we should try for a unified approach to things.

**Pull request merged**